### PR TITLE
Add missing tracepoint components and groups

### PR DESCRIPTION
**Pull request by pshipton**
See https://github.com/eclipse-openj9/openj9/issues/20175

### CODE REVIEW

**Review by dmitripivkine — APPROVED** [review]

File: docs/xtrace.md [hunk 1]
```diff
@@ -261,9 +278,15 @@ The following table lists all the tracepoint groups (`<group>`). Each group is a
 |Component name or names     |  Group name       |  Description                                                                                                              |
 |----------------------------|-------------------|---------------------------------------------------------------------------------------------------------------------------|
 |**j9mm**                    | `gclogger`        | A set of tracepoints that record each garbage collection cycle. Equivalent to `-verbose:gc` output                        |
```
Comment (keithc-ca) [inline docs/xtrace.md:280]: This also applies to `omrmm`. The list of groups is incomplete; I found this list (in OMR and OpenJ9):
```
	adaptivethread
	ageblending
	aging
	allocate
	alloclarge
	allocthreshold
	arraylet
	bytecodeMethods
	cardscrubbing
	checkjni
	checkvm
	classinit
	compact
	compiledMethods
	concurrent
	copyforwardscheme
	cuda
	darkMatterComparison
	dispatcher
	double_map
	dynamiccollectionset
	exvmaccess
	file
	gclogger
	heap
	j9file
	j9filestrean
	j9gs
	j9heap
	j9hypervisor
	j9mem
	j9mmap
	j9ri
	j9shared
	j9shmem
	j9shsem
	j9sig
	j9signal
	j9sl
	j9sock
	j9sysinfo
	j9syslog
	j9virt_rwlock
	j9vmem
	kickoff
	loaresize
	markdelegate
	mem
	methodArguments
	mmap
	nativeMethods
	nlsmessage
	omrfilestream
	omrshared
	omrshmem
	omrshsem
	omrsock
	oracle
	OSCache
	parallel
	percolate
	perfmon
	reclaim
	rememberedset
	resize
	runtimeexec
	scavenge
	scavenger
	signal
	sl
	stackremember
	staticMethods
	stringtable
	sysinfo
	syslog
	tarok
	testset1
	throw
	unload
	verboseclass
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Comment by pshipton** [comment]
@dmitripivkine I've added many more GC groups, can you pls check over the descriptions and see if any should be changed. 

Of interest, there are a few tracepoints in group `scavenge` but more in group `scavenger`, I wonder if there is a typo.

**Comment by pshipton** [comment]
I left out some old j9 port library groups where all the tracepoints are obsolete.

**Comment by dmitripivkine** [comment]
@amicic Do we need to add more j9mm groups?

**Comment by amicic** [comment]
These j9mm have been identified, by not added to the list 

kickoff
loaresize
markdelegate

**Comment by pshipton** [comment]
Added the missing j9mm groups, and others.

I've also opened https://github.com/eclipse/omr/pull/7470 to fix "j9filestrean", which should be "omrfilestream".

File: docs/xtrace.md [hunk 2]
```diff
@@ -241,35 +246,108 @@ The following table lists the possible Java components (`<component>`). To inclu
...
-|**mt**                      | `compiledMethods` | A set of tracepoints that record compiled Java methods.                                                                   |
-|**mt**                      | `nativeMethods`   | A set of tracepoints that record Java native methods.                                                                     |
-|**mt**                      | `staticMethods`   | A set of tracepoints that record Java static methods.                                                                     |
+|Component name or names     |  Group name            |  Description                                                                                                              |
+|----------------------------|------------------------|---------------------------------------------------------------------------------------------------------------------------|
+|**j9mm**                    | `adaptivethread`       | A set of tracepoints that record GC adaptive threading.                                                                   |
```
Comment (keithc-ca) [inline docs/xtrace.md:280]: In `j9prt.tdf`, I found one group not listed here:
```
TraceExit=Trc_PRT_j9hypervisor_vendor_init_exit Group=j9mem Overhead=1 Level=5 NoEnv Template="j9hypervisor_vendor_init exited with ret=%d"
```
That tracepoint is used in `j9hypervisor_common.c`.

Comment (pshipton) [reply docs/xtrace.md:280]: How about I change the group for this tracepoint to j9hypervisor, it seems an error for it to be j9mem. Also a group with a single tracepoint isn't that useful.

Comment (pshipton) [reply docs/xtrace.md:280]: https://github.com/eclipse-openj9/openj9/pull/20220

Comment (keithc-ca) [reply docs/xtrace.md:280]: Agreed.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins doc stage

**Review by dmitripivkine — APPROVED** [review]

**Pull request merged**